src/Build/Logging/BinaryLogger/BinaryLogger.cs

```diff
@@ -239,7 +239,14 @@ public void Shutdown()
                     {
                         using (FileStream fileStream = File.OpenRead(archiveFilePath))
                         {
-                            eventArgsWriter.WriteBlob(BinaryLogRecordKind.ProjectImportArchive, fileStream);
+                            if (fileStream.Length > int.MaxValue)
+                            {
+                                LogMessage("Imported files archive exceeded 2GB limit and it's not embedded.");
```

Comment:
It looks like these messages are expected to have the `Key=Value` format. Maybe
```suggestion
                                LogMessage("Error=Imported files archive exceeded 2GB limit and it's not embedded.");
```
?

Reply:
Yes, there will be parsing of key/value with `=` as delimiter. I wanted to wait for @KirillOsenkov to provide advice.

Reply:
I had to change template, so I'm adding screenshot here to keep the discussion context:

I'm not sure if the message about 2GB limit is necessary. If so, the formatting doesn't look great.
![image](https://github.com/dotnet/msbuild/assets/12775388/2960bbec-daa5-436c-9265-78d453c3d51b)

Reply:
Sorry, why do we need the = here?

Reply:
The viewer appears to be expecting messages formatted this way. If it doesn't find `=` in the message it treats the full string as a 'key' with an empty 'value', see the trailing `=` in @MichalPavlik's screenshot.

Reply:
Let's fix the viewer then. If it doesn't find the = it should just interpret as a simple message. I don't remember why it behaves this way currently.

Reply:
Yes, let's remove the = and I will fix the viewer to handle this correctly.

Reply:
I suspect it's falling through into this branch:
https://github.com/KirillOsenkov/MSBuildStructuredLog/blob/7d6b5c293368668e2ef20183c475c6a9f4552bec/src/StructuredLogger/Construction/MessageProcessor.cs#L132-L137

Should check to only fall through if the message contains a `=`

Reply:
> Let's fix the viewer then. If it doesn't find the = it should just interpret as a simple message. I don't remember why it behaves this way currently.

I think the key=value parsing logic is there because we were emitting all (2) messages in this format. Having possibility to emit just plain message would be very nice. @ladipro, I don't think this PR should depend on the parsing logic of binlog viewer. Please resolve the comment if you are satisfied.

Reply:
Sounds great, resolving.